### PR TITLE
Move installation.rst to installation/*.rst

### PR DESCRIPTION
**Pull request by aclark4life**
- Add index
- Move content to sections
- Rearrange sections

Changes proposed in this pull request:

 * Adds navigation to Installation section
 * Rearrange sections to prioritize basic installation, followed by supported platforms and Python versions, followed by building from source and old releases
Screenshot 2024-02-24 at 6 36 01 PM

### CODE REVIEW

**Comment by hugovk** [comment]
My biggest concern is linkrot, existing links out will break and result in 404s.

For example, answers on Stack Overflow will no longer link to the correct place.

**Comment by aclark4life** [comment]
@hugovk Good call, I've added a95ca255f7d886831d94dcb6f2fab7123b6ad610 to address that issue.

File: docs/conf.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "sphinx_inline_tabs",
     "sphinx_removed_in",
     "sphinxext.opengraph",
+    "sphinx_reredirects",
```
Comment (radarhere) [inline docs/conf.py:39]:
```suggestion
    "sphinx_reredirects",
    "sphinxext.opengraph",
```
Alphabetic order.

**Comment by hugovk** [comment]
Is there a way to redirect links like this?

https://pillow.readthedocs.io/en/stable/installation.html#building-from-source

With the preview:

https://pillow--7832.org.readthedocs.build/en/7832/installation.html#building-from-source

It now only goes to the ToC:

https://pillow--7832.org.readthedocs.build/en/7832/installation/index.html

**Comment by aclark4life** [comment]
Alternatively, we could leave note, something like:

Screenshot 2024-02-27 at 7 11 21 AM

File: docs/installation.rst [hunk 2]
```diff
@@ -1,597 +1,27 @@
 Installation
```
Comment (radarhere) [inline docs/installation.rst:1]:
```suggestion
:orphan:

Installation
```
Silences the ["document isn't included in any toctree"](https://github.com/python-pillow/Pillow/actions/runs/8107308747/job/22158625064#step:8:147) warning.

**Comment by aclark4life** [comment]
@hugovk @radarhere OK to merge this one? 🤔

File: docs/conf.py [hunk 3]
```diff
@@ -350,3 +351,6 @@
     "pillow-logo-dark-text-1280x640.png"
 )
 ogp_image_alt = "Pillow"
+
+# sphinx-reredirects
+# redirects = {"installation.html": "installation/index.html"}
```
Comment (radarhere) [inline docs/conf.py:356]: If this is commented out, then we aren't actually using sphinx-reredirects, right? We could remove it, like https://github.com/radarhere/Pillow/commit/7aba72a48f1b55a3716409f66e772af740f40616?

**Comment by aclark4life** [comment]
> @hugovk @radarhere OK to merge this one? 🤔

Hold on to your butts

**Comment by hugovk** [comment]
Sorry for the late reply!

I think this is mostly fine, the original page was very large and it definitely looks much better for most of these to be their own page. 👍 

I've opened https://github.com/python-pillow/Pillow/pull/7858 with some little tweaks.

**Pull request merged**